lib/gearbox.ex

```diff
@@ -232,6 +234,11 @@ defmodule Gearbox do
     end
   end
 
+  defp maybe_apply_changeset_changes(%Ecto.Changeset{} = changeset),
+    do: Ecto.Changeset.apply_changes(changeset)
```

Comment:
references should Ecto is optional so should only be in `gearbox/ecto.ex`

Reply:
Thanks for your feedback, it totally makes sense! All addressed!